### DISCUSSION TITLE
document_exporter with docker and crontab

### DISCUSSION THREAD

**Discussion by martinpichlo** [jonaswinkler/paperless-ng#903] [Support] [answered]
Hi,

I'm using paperless-ng within docker. I would like to do daily backups via document_exporter. Therefore I have created a script starting the backup. This is working fine, as long as I start it manually. But it just do nothing if it is starting via crontab, but the script is running, because my protocol is written.

What I did:

1. Step: creating a backup script
```
#!/bin/sh
cd /home/martin/paperless-ng
#docker-compose exec webserver document_exporter ../export
#docker-compose exec webserver python manage.py document_exporter /usr/src/paperless/export
docker-compose exec webserver sh -c "python manage.py document_exporter /usr/src/paperless/export"
date >> protokoll.log
```

2. Step: maintain crontab -e for user martin
```
# min   hour    day     month   weekday command
02       17       *       *       *      ~/backup_script/backup.sh
```
What am I doing wrong?

PS: Is there a need to stop anything before starting the backup?

**Comment by sbrunner** [comment]
There isn't ans cron demon in the container, personally I add some `docker exec ...` or `docker-compose exec ...` in the cron tab of the host (out of Docker)

**Reply by martinpichlo** [reply]
Forgot to say it. I already start the command from the host.

**Comment by jonaswinkler** [comment]
> docker-compose exec webserver sh -c "python manage.py document_exporter /usr/src/paperless/export"

I'm pretty sure you can simplify that to

`docker-compose exec webserver document_exporter /usr/src/paperless/export`

**Reply by martinpichlo** [reply]
Thank you for pointing this out. I found several variants (see my script above). May something to add in the documentation?

**Reply by jonaswinkler** [reply]
The form I posted is in the documentation and the recommended way to execute management commands with docker.

**Comment by martinpichlo** [comment, marked as answer]
I still have no clue whats going on, but I have a solution. But step by step.

First I adjust the docker-compose command and added absolut pathes. But as before the document_exporter is working, when calling the script manually. But calling it via crontab -e nothing happens.

`/usr/bin/docker-compose exec webserver document_exporter /usr/src/paperless/export`

After some searching I found the docker-compose issue [2293](https://github.com/docker/compose/issues/2293).One of the answers there is, to use the "-T" option. I haven't found nothing about this option, but it works like a charm.

`/usr/bin/docker-compose exec -T webserver document_exporter /usr/src/paperless/export`

**Reply by jonaswinkler** [reply]
```
    -T                Disable pseudo-tty allocation. By default `docker-compose exec`
                      allocates a TTY.
```

This works probably because the management commands tries to interact with the TTY.

**Reply by martinpichlo** [reply]
Thanks

**Comment by martinpichlo** [comment]
https://github.com/jonaswinkler/paperless-ng/issues/923 :-)
